getting-started/interact-with-contract.md

```diff
@@ -0,0 +1,184 @@
+wasmcli query wasm contract-state all $CONTRACT | jq -r '.[0].val' | base64 -d
+
+# or try a "smart query", executing against the contract
+wasmcli query wasm contract-state smart $CONTRACT '{}'
+# (since we didn't implement any valid QueryMsg, we just get a parse error back)
+```
+
+Once we have the funds in the escrow, let us try to release them. First, failing to do so with a key that is not the verifier, then using the proper key to release.
+
+```bash
+# execute fails if wrong person
+APPROVE='{"approve":{"quantity":[{"amount":"50000","denom":"ucosm"}]}}'
+wasmcli tx wasm execute $CONTRACT "$APPROVE" --from thief -y
+# looking at the logs should show: "execute wasm contract failed: Unauthorized"
+# and bob should still be broke (and broken showing the account does not exist Error)
+wasmcli query account $(wasmcli keys show bob -a)
+
+# but succeeds when fred tries
+wasmcli tx wasm execute $CONTRACT "$APPROVE" --from fred -y
+wasmcli query account $(wasmcli keys show bob -a)
+wasmcli query account $CONTRACT
+```
+
+## Node Console
+
+If you set up the Node Console / REPL in the
+[client setup section](./using-the-sdk), you can use that
+to deploy and execute your contract.
+I think you will find that JSON manipulation and parsing
+is a bit nicer in JavaScript than in Shell Script.
+
+First, go to the cli directory and start up your console:
+
+```sh
+npx @cosmjs/cli --init https://raw.githubusercontent.com/CosmWasm/cosmjs/v0.22.0/packages/cli/examples/helpers.ts
+```
+
+Now, we make all the keys and initialize clients:
+
+```js
+const fredSeed = loadOrCreateMnemonic("fred.key");
+const {address: fredAddr, client: fredClient} = await connect(fredSeed, {});
+
+// bob doesn't have a client here as we will not
+// submit any tx with this account, just query balance
+const bobSeed = loadOrCreateMnemonic("bob.key");
+const bobAddr = await mnemonicToAddress("cosmos", bobSeed);
+
+const thiefSeed = loadOrCreateMnemonic("thief.key");
+
+const {address: thiefAddr, client: thiefClient} = await connect(thiefSeed, {});
+
+console.log(fredAddr, bobAddr, thiefAddr);
+```
+
+Hit the faucet it needed for fred , so he has tokens to submit transactions:
+
+```js
+fredClient.getAccount();
+// if "undefined", do the following
+hitFaucet(defaultFaucetUrl, fredAddr, "COSM")
+fredClient.getAccount();
+
+thiefClient.getAccount();
+// if "undefined", do the following
+hitFaucet(defaultFaucetUrl, thiefAddr, "COSM")
+thiefClient.getAccount();
+
+// check bobAddr has no funds
+fredClient.getAccount(bobAddr);
+
+// get the working directory (needed later)
+process.cwd()
+```
+
+### Uploading with JS
+
+Now, we go back to the Node console and upload the
+contract and instantiate it:
+
+```js
+const wasm = fs.readFileSync('contract.wasm');
+// fake source, but this can be verified to be false
+// by any careful observer
+const up = await fredClient.upload(wasm, { source: "https://crates.io/api/v1/crates/cw-escrow/0.4.0/download", builder: "cosmwasm/rust-optimizer:0.9.0"});
```

Comment:
https://crates.io/api/v1/crates/cw-escrow/0.6.0/download does not exist, we need to upload it @ethanfrey